test/stdlib/Mirror.swift

```diff
@@ -921,6 +921,72 @@ mirrors.test("class/Cluster") {
 //===--- Miscellaneous ----------------------------------------------------===//
 //===----------------------------------------------------------------------===//
 
+protocol Box<Value> {
+  associatedtype Value
+  var value: Value {get}
+}
+
+mirrors.test("Extended Existential (struct)") {
+  struct Container<Value>: Box {
+    var value: Value
+  }
+  func genericErase<T>(_ value: T) -> Any {
+    value
+  }
+  let container: any Box<Int> = Container(value: 42)
+  if #available(macOS 13, *) {
+    let subject = genericErase(container)
+    let mirror = Mirror(reflecting: subject)
+    let children = mirror.children
+    expectEqual(1, children.count)
+    let first = children.first!
+    expectEqual("value", first.label)
+    expectEqual(42, first.value as! Int)
+  }
+}
+
+protocol OBox<Value>: AnyObject {
+  associatedtype Value
+  var value: Value {get}
+}
+
+mirrors.test("Extended Existential (class)") {
+  class Container<Value>: OBox {
+    var value: Value
+    init(value: Value) { self.value = value }
+  }
+  func genericErase<T>(_ value: T) -> Any {
+    value
+  }
+  let container: any OBox<Int> = Container(value: 42)
+  if #available(macOS 13, *) {
+    let subject = genericErase(container)
+    let mirror = Mirror(reflecting: subject)
+    let children = mirror.children
+    expectEqual(1, children.count)
+    let first = children.first!
+    expectEqual("value", first.label)
+    expectEqual(42, first.value as! Int)
+  }
+}
+
+mirrors.test("Extended Existential (metatype)") {
```

Comment:
@slavapestov Can you think of a better way to verify the mirror support here?   I have checked in a debugger that it does actually hit the expected case inside the Mirror implementation.